### ISSUE TITLE
Audio Time-Lag

### ISSUE THREAD

**Issue by FA-Bubba** [SamDel/ChromeCast-Desktop-Audio-Streamer#17]
REVISED...   See attached comment
-------------------------------------------
I downloaded and installed audio-cast,exe on my Windows 7 Pro PC (16 GB RAM).  Audio-cast sees my Google devices (2 minis, a hub, 1 Chromecast Audio & 1 Chromecast).  I selected the 2 Minis & the Chromecast Audio, and heard the connect sound from each speaker.  I started a Playlist in iTunes, and it played to all of the speakers, but there a time-lag in the sound stream.  That is, the speakers are not sync'd, but output the sound with various delays.  The delay can be minor, or significant, with no consistency...

What tools &/or actions can I try to get the sound output to all speakers on the same time-line?

Thank you!

**Comment by FA-Bubba** [comment]
My post above was from an older version (0.3) that I found a few days ago.  I eventually found this version by SamDel (v 1.3) that has a more finished UI, and seems to work very well...  I am not noticing lags between the Google speakers, but the audio does seem to be buffered from the PC, so there is a significant lag between the PC and the Audio Stream from the Google speakers...

Turning the PC speakers OFF was a quick fix, but it would be nice to eliminate the lag altogether, if that is possible.

GREAT App, SamDel... THANK YOU for your App -- I now have multi-room music !!!

**Comment by SamDel** [comment]
You're talking about two different projects :). Audio-cast.exe is the project by [Acidhax](https://github.com/acidhax/chromecast-audio-stream), which has a complete different codebase. Both projects work basically the same.

Syncing devices is not supported by Chromecast, but two devices can be (accidentally) in sync of course. You can also try the trick I described [here](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/issues/9#issuecomment-378016490).

There will always be a lag between the PC and the Chromecast, that's mainly the buffer on the Chromecast. You can reduce the lag with the (experimental) lag control, but the sound quality will degrade.

**Comment by FA-Bubba** [comment]
Thank you for the follow-up!...
I eventually surmised they were two different solutions.  Yours seems much more evolved, and I have been using it since I discovered it -- KUDOS for your coding!!!

Earlier today, I realized that the speakers I have set up (6) are not always in sync; that's not really a big issue, except where two speakers are in the same room (something I will change later this week).  I did note that stopping iTunes, waiting a short while, then restarting it does seem to get the speakers "closer" in sync...

FYI, on Friday, I was in a long discussion with Google Home Support, when I asked them how to cast from PC Audio to Google Home Speakers -- they thought the Desktop could be cast via the Chrome browser, but nothing they asked me to try worked, and they eventually concluded it was not possible with existing Google software.  At the time, I was only aware of the Acidhax, App, and suggested they take a look at that.  They said they were interested in a solution like that -- should I redirect them to your solution?  (They seem to open to suggestions for Google Home solutions:   They are working on a different idea I suggested to them a few weeks ago.)

In any event, THANK YOU for your great work!

**Comment by SamDel** [comment]
Stopping playback is also a nice trick! I didn't think of it but then the buffers on the devices are also (most likely) more in line with each other (because when you don't play music nothing is streamed to the devices).

It would be nice if Google makes something that works out of the box, so please continue to inspire them!

**Comment by FA-Bubba** [comment]
A new discovery last night:  I was using your Desktop Audio App to cast to 5 speakers in different rooms.  The PC was upstairs, and I was downstairs -- I went to two speakers and said "Hey Google, Stop Casting", and those speakers each stopped.  Later, when I went upstairs, I checked the App, and it showed those two speakers as OFF, and I was able to resume casting by simply clicking each device name.   But it did not work the other way -- I went to a speaker I had silenced above, and said "Hey Google, Resume Casting"; the speaker responded "Nothing is playing right now".
